### ISSUE TITLE
ImportError: libDPCTLSyclInterface.so.0: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by krzeszew** [IntelPython/numba-dppy#1505]
Hi,

I have problems when importing `numba_dpex` using Python 3.11 on Ubuntu 22.04. I've installed latest numba-dpex==0.23.0 using pip. When importing package there is the following error:

```
>>> import numba_dpex
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ula/.local/lib/python3.10/site-packages/numba_dpex/__init__.py", line 19, in <module>
    from .kernel_api_impl.spirv import target as spirv_kernel_target
  File "/home/ula/.local/lib/python3.10/site-packages/numba_dpex/kernel_api_impl/spirv/target.py", line 23, in <module>
    from numba_dpex.core.datamodel.models import _init_kernel_data_model_manager
  File "/home/ula/.local/lib/python3.10/site-packages/numba_dpex/core/__init__.py", line 6, in <module>
    from .types import *
  File "/home/ula/.local/lib/python3.10/site-packages/numba_dpex/core/types/__init__.py", line 5, in <module>
    from .dpctl_types import DpctlSyclEvent, DpctlSyclQueue
  File "/home/ula/.local/lib/python3.10/site-packages/numba_dpex/core/types/dpctl_types.py", line 13, in <module>
    from numba_dpex.core.runtime import context as dpexrt
  File "/home/ula/.local/lib/python3.10/site-packages/numba_dpex/core/runtime/__init__.py", line 7, in <module>
    from ._dpexrt_python import c_helpers
ImportError: libDPCTLSyclInterface.so.0: cannot open shared object file: No such file or directory
```

<details close>
  <summary>pip list</summary>

```
Package               Version
--------------------- ----------------
apturl                0.5.2
attrs                 21.2.0
Automat               20.2.0
Babel                 2.8.0
bcrypt                3.2.0
blinker               1.4
Brlapi                0.8.3
certifi               2020.6.20
chardet               4.0.0
click                 8.0.3
cloud-init            24.2
colorama              0.4.4
command-not-found     0.3
configobj             5.0.6
constantly            15.1.0
cryptography          3.4.8
cupshelpers           1.0
dbus-python           1.2.18
defer                 1.0.6
distro                1.7.0
distro-info           1.1+ubuntu0.2
dpcpp-cpp-rt          2025.0.3
dpcpp-llvm-spirv      2024.2.0
dpctl                 0.18.1
dpnp                  0.16.0
duplicity             0.8.21
fasteners             0.14.1
future                0.18.2
httplib2              0.20.2
hyperlink             21.0.0
idna                  3.3
importlib-metadata    4.6.4
incremental           21.3.0
iniconfig             1.1.1
intel-cmplr-lib-rt    2025.0.3
intel-cmplr-lib-ur    2025.0.3
intel-cmplr-lic-rt    2025.0.3
intel-opencl-rt       2025.0.3
intel-openmp          2025.0.3
intel-sycl-rt         2025.0.3
jeepney               0.7.1
Jinja2                3.0.3
jsonpatch             1.32
jsonpointer           2.0
jsonschema            3.2.0
keyring               23.5.0
language-selector     0.1
launchpadlib          1.10.16
lazr.restfulclient    0.14.4
lazr.uri              1.0.6
libvirt-python        8.0.0
llvmlite              0.43.0
lockfile              0.12.2
louis                 3.20.0
macaroonbakery        1.3.1
Mako                  1.1.3
MarkupSafe            2.0.1
mkl                   2025.0.1
monotonic             1.6
more-itertools        8.10.0
netifaces             0.11.0
numba                 0.60.0
numba-dpex            0.23.0
numpy                 2.0.2
oauthlib              3.2.0
olefile               0.46
onemkl-sycl-blas      2025.0.1
onemkl-sycl-dft       2025.0.1
onemkl-sycl-lapack    2025.0.1
onemkl-sycl-rng       2025.0.1
onemkl-sycl-stats     2025.0.1
onemkl-sycl-vm        2025.0.1
packaging             21.3
paramiko              2.9.3
pexpect               4.8.0
Pillow                9.0.1
pip                   22.0.2
pluggy                0.13.0
protobuf              3.12.4
ptyprocess            0.7.0
py                    1.10.0
pyasn1                0.4.8
pyasn1-modules        0.2.1
pycairo               1.20.1
pycups                2.0.1
Pygments              2.11.2
PyGObject             3.42.1
PyHamcrest            2.0.2
PyJWT                 2.3.0
pymacaroons           0.13.0
PyNaCl                1.5.0
pyOpenSSL             21.0.0
pyparsing             2.4.7
pyRFC3339             1.1
pyrsistent            0.18.1
pyserial              3.5
pytest                6.2.5
python-apt            2.4.0+ubuntu4
python-dateutil       2.8.1
python-debian         0.1.43+ubuntu1.1
python-magic          0.4.24
pytz                  2022.1
pyxdg                 0.27
PyYAML                5.4.1
reportlab             3.6.8
requests              2.25.1
SecretStorage         3.3.1
service-identity      18.1.0
setuptools            59.6.0
six                   1.16.0
sos                   4.5.6
ssh-import-id         5.11
tbb                   2022.0.0
tcmlib                1.2.0
toml                  0.10.2
Twisted               22.1.0
ubuntu-drivers-common 0.0.0
ubuntu-pro-client     8001
ufw                   0.36.1
umf                   0.9.1
unattended-upgrades   0.1
urllib3               1.26.5
usb-creator           0.3.7
wadllib               1.3.6
wheel                 0.37.1
xdg                   5
xkit                  0.0.0
zipp                  1.0.0
zope.interface        5.4.0
```

</details>

When I was trying to find it using `find` function, there wasn't any found. What cloud I possibly do to fix this problem.

Thank you for your help!

**Comment by Ankur-singh** [comment]
I'm also facing the same issue.

**Comment by ZzEeKkAa** [comment]
Thank you for reporting the issue.

I guess there are some design problems when using pip. Try using something like `export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$VIRTUAL_ENV/lib`. Target location may be different depending on where `libDPCTLSyclInterface.so.0` is located. To find it use `find $VIRTUAL_ENV | grep libDPCTLSyclInterface`.

@oleksandr-pavlyk may have better way of fixing the issue.

**Comment by ZzEeKkAa** [comment]
As I recall, we've been updating `libDPCTLSyclInterface` library in dpctl to remove duplicates. Try to rollback to previous version of dpctl.

**Comment by oleksandr-pavlyk** [comment]
I am trying to reproduce, but in your report you state creating environment with Python 3.11, while exception backtrace reveals 3.10 being used.

**Comment by oleksandr-pavlyk** [comment]
I can reproduce the reported `ImportError` exception. 

<details>
<summary>
Steps executed to reproduce
</summary>

## Create conda environment with only CPython and package manager

```bash
conda create -n py311 -c conda-forge --override-channels python=3.11 pip
conda activate py311
```

## Install numba-dpex using `pip` into this environment

```bash
pip install --no-cache-dir numba-dpex==0.23.0
```

## Reproducing  `ImportError`

```bash
(py311) opavlyk@opavlyk-mobl:~$ python -c "import numba_dpex; print(numba_dpex.__version__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/opavlyk/mamba/envs/py311/lib/python3.11/site-packages/numba_dpex/__init__.py", line 19, in <module>
    from .kernel_api_impl.spirv import target as spirv_kernel_target
  File "/home/opavlyk/mamba/envs/py311/lib/python3.11/site-packages/numba_dpex/kernel_api_impl/spirv/target.py", line 23, in <module>
    from numba_dpex.core.datamodel.models import _init_kernel_data_model_manager
  File "/home/opavlyk/mamba/envs/py311/lib/python3.11/site-packages/numba_dpex/core/__init__.py", line 6, in <module>
    from .types import *
  File "/home/opavlyk/mamba/envs/py311/lib/python3.11/site-packages/numba_dpex/core/types/__init__.py", line 5, in <module>
    from .dpctl_types import DpctlSyclEvent, DpctlSyclQueue
  File "/home/opavlyk/mamba/envs/py311/lib/python3.11/site-packages/numba_dpex/core/types/dpctl_types.py", line 13, in <module>
    from numba_dpex.core.runtime import context as dpexrt
  File "/home/opavlyk/mamba/envs/py311/lib/python3.11/site-packages/numba_dpex/core/runtime/__init__.py", line 7, in <module>
    from ._dpexrt_python import c_helpers
ImportError: libDPCTLSyclInterface.so.0: cannot open shared object file: No such file or directory
```

</details>

## Reason for exception

`numba-dpex` is presently in the maintenance mode. Its last release, version 0.23.0, was published in June 2024, concurrently with Intel Python 2024.2.0 release, containing `dpctl==0.17.0` and DPC++ 2024.2.0 runtime libraries. 

Intel Python 2025.0.0 release updates `dpctl` to 0.18.* and drops use of so-versioning for DPCTLSyclInterface library (https://github.com/IntelPython/dpctl/pull/1773). 

The native extension extension for `numba_dpex._dpexrt_python` needs to be rebuilt against this newer version of `dpctl`, but `numba_dpex` was not included in 2025.0 release (see section 2025.0 in https://www.intel.com/content/www/us/en/developer/articles/release-notes/distribution-for-python-release-notes.html) hence the update was not published.

## Resolutions

1. You can install latest build from dppy/label/dev Anaconda channel/index (https://anaconda.org/dppy/numba-dpex)
2. You can downgrade to `dpctl==0.17.0` and ensuring that DPC++ runtime libraries 2024.2.0 are installed

**Comment by krzeszew** [comment]
Thank you for the help. It works with 2024.2 DPC++ runtime libraries.
